File: php/mediator/Finacial.php
<?php


class Finacial implements Department
{
    // 持有中介者（总经理）的引用
    private $mediator;

    public function __construct($m)
    {
        $m->register('finacial', $this);
        $this->mediator = $m;
    }

    /**
     * 做部门的事情
     * @return [type] [description]
     */
    public function selfAction()
    {
        echo '我在整理财务报表!', PHP_EOL;
    }

    /**
     * 向总经理发送申请
     * @return [type] [description]
     */
    public function outAction()
    {
        echo '汇报工作！资金充足，可以考虑新的方向', PHP_EOL;
    }
}

File: java/pattern/src/mediator/Finacial.java
package mediator;

public class Finacial implements Department {
    // 持有中介者（总经理）的引用
    private Mediator m;

    public Finacial(Mediator m) {
        this.m = m;
        m.register("finacial", this);
    }

    @Override
    public void selfAction() {
        System.out.println("我在整理财务报表！");
    }

    @Override
    public void outAction() {
        System.out.println("汇报工作！资金充足，可以考虑新的方向");
    }
}
